### PR TITLE
Runnable software to interact with magnetometer and experiment system

### PR DESCRIPTION
**Pull request by yotam180**
What's here?

* Interacting with the magnetometer sensor
* Finding the lab address
* Continuously sending data from magnetometer to the lab

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    else:
+        axes = [0, 0, 0]
+        magnitude = 0
+
+    print("Updating DataHub")
+    hub.set({"sat_mag": [axes, magnitude]})
+    print("Updated DataHub")
+
+    time.sleep(0.1)
```
Comment (ron-popov) [inline main.py]: The refresh rate of the magnetometer is 8 times a second, trying to read magnemeter reading 10 times a second could give you some unexpected results, i suggest changing that to 0.125 seconds.

File: main.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    from Modules.Magnetorquer.hBridge import hBridge
+
+from Modules.DataHub.remote import DataHub
+
+import time
+import math
+
+if SAT:
+    sensor = magnetometer_factory.initialize()
+    actuator = hBridge(23, 24, 14, 15)
```
Comment (ron-popov) [inline main.py]: Why aren't you using mangetorquer ?

File: main.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    else:
+        axes = [0, 0, 0]
+        magnitude = 0
+
+    print("Updating DataHub")
+    hub.set({"sat_mag": [axes, magnitude], "sat_mag_ping": time.time()})
+    print("Updated DataHub")
+
+    time.sleep(0.1)
```
Comment (ron-popov) [inline main.py]: The refresh rate of the mangetometer is 8 times a second, if you try to read data from it 10 times a second, things could get nasty os i would suggest changing that to 0.125

**Review by ron-popov — CHANGES_REQUESTED** [review]

File: main.py (hunk 3 above)
Comment (yotam180) [reply main.py]: I actually planned ` hub.set({"sat_mag": [axes, magnitude], "sat_mag_ping": time.time()})` to take additional ~0.1-0.2 sec. Do you really think this is necessary to validate it at the `sleep` level?

File: main.py (hunk 2 above)
Comment (yotam180) [reply main.py]: He, I wanted to but got stuck while trying to think of a way to implement those two together with the lab system. For now I think I will just remove it (Or rethink about it). Thanks for pointing that out :sunglasses:

**Review by ron-popov — APPROVED** [review]
In Yotam We Trust

**Comment by yotam180** [comment]
Thank you :)

**Pull request merged**